src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Commands/Upgrade/ConsoleUpgrade.cs

```diff
@@ -43,6 +51,8 @@ public class ConsoleUpgrade : IAppCommand
             _telemetry = telemetry ?? throw new ArgumentNullException(nameof(telemetry));
             _stateManager = stateManager ?? throw new ArgumentNullException(nameof(stateManager));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _writerProvider = writerProvider;
```

Comment:
Either have to `?? throw new ArgumentNullException(nameof(writerProvider));` this or check null & ignore down in `WriteUpgradeReport`